### DISCUSSION TITLE
[Problem] symbol auto name

### DISCUSSION THREAD

**Discussion by HaoLiuHust** [apache/incubator-mxnet#14741] [Q&A]
mxnet will give a symbol a name automatically if I do not give it one, for example a "+" operator and so on, it works well when I construct a network symbol from scratch, but when I get a symbol from symbol-json or checkpoint, and then concat some other sym without name, there will be variable with same name. Is it by design or a bug?

**Comment by HaoLiuHust** [comment]
finally I found a solution, use NameManager

**Comment by frankfliu** [comment]
@mxnet-label-bot add [question]
